frontend/src/pages/patients/PatientsPage.tsx

```diff
@@ -65,9 +71,101 @@ function patientAliasFilterWhereVariables(value: string): PatientAliasWhere[] {
   }
 }
 
-export const PatientsPage: React.FunctionComponent = (props) => {
+export default function PatientsPage({ setUserEmail }: { setUserEmail: any }) {
   const params = useParams();
 
+  const [searchWithMRNs, setSearchWithMRNs] = useState(false);
+  const [patientIdsTriplets, setPatientIdsTriplets] = useState<
+    PatientIdsTriplet[]
+  >([]);
+  const [patientsListColumns, setPatientsListColumns] =
+    useState(PatientsListColumns);
+  const [popupOpen, setPopupOpen] = useState(false);
+
+  useEffect(() => {
+    function handleLogin(event: any) {
+      if (event.origin !== "http://localhost:4001") return;
+      setUserEmail(event.data);
+      setPopupOpen(false);
+    }
+
+    if (popupOpen) {
+      window.addEventListener("message", handleLogin);
+      return () => {
+        window.removeEventListener("message", handleLogin);
+      };
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [popupOpen]);
+
+  useEffect(() => {
+    if (searchWithMRNs && patientIdsTriplets.length > 0) {
+      const colsWithMRNs = PatientsListColumns.map((column) => {
+        if (column.headerName === "Patient MRN") {
+          return {
+            ...column,
+            hide: false,
+            valueGetter: (params: any) => {
+              const cmoId = params.data.value;
+              const patientIdsTriplet = patientIdsTriplets.find(
+                (triplet) => triplet.cmoId === cmoId
+              );
+              if (patientIdsTriplet) {
+                return patientIdsTriplet.ptMrn;
+              } else {
+                return "";
+              }
+            },
+          };
+        } else {
+          return column;
+        }
+      });
+      setPatientsListColumns(colsWithMRNs);
+    } else {
+      setPatientsListColumns(PatientsListColumns);
+    }
+  }, [searchWithMRNs, patientIdsTriplets]);
```

Comment:
In MRN search mode, this `useEffect` helps show and populate a new column with MRN values that correspond to CMO and DMP Patient IDs.